pandas/tests/series/test_logical_ops.py

```diff
@@ -264,44 +264,27 @@ def test_logical_ops_with_index(self, op):
         result = op(ser, idx2)
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.filterwarnings("ignore:passing object-dtype arraylike:FutureWarning")
-    def test_reversed_xor_with_index_returns_index(self):
-        # GH#22092, GH#19792
+    def test_reversed_xor_with_index_returns_series(self):
+        # GH#22092, GH#19792 pre-2.0 these were aliased to setops
         ser = Series([True, True, False, False])
         idx1 = Index(
             [True, False, True, False], dtype=object
         )  # TODO: raises if bool-dtype
         idx2 = Index([1, 0, 1, 0])
 
-        msg = "operating as a set operation"
-
-        expected = Index.symmetric_difference(idx1, ser)
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            result = idx1 ^ ser
-        tm.assert_index_equal(result, expected)
+        expected = ser ^ idx1.values
```

Comment:
IMO would be safer if `expected` is a concrete series rather than result of a calculation using xor (that you're aslo testing if it works correctly.

Likewise in other similar locations.